packages/front-end/components/Settings/UpgradeModal/SelfHostedUpgradeForm.tsx

```diff
@@ -27,25 +29,89 @@ export default function SelfHostedUpgradeForm({
 
   const { accountPlan, name, email, organization } = useUser();
 
+  const customerContext = {
+    organizationCreated: organization.dateCreated,
+    currentSeats: getNumberOfUniqueMembersAndInvites(organization),
+    currentPlan: accountPlan,
+    currentBuild: getGrowthBookBuild(),
+    ctaSource: source,
+  };
+
   useEffect(() => {
-    track("View Upgrade Modal", {
-      accountPlan,
-      source,
-    });
+    track("Self Hosted: View Upgrade Modal", customerContext);
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
   const form = useForm({
     defaultValues: {
-      name,
-      email,
-      seats: Math.max(organization?.members?.length || 10, 10),
       companyName: organization?.name,
       organizationId: organization?.id,
-      plan: "pro",
+      name,
+      email,
+      context: customerContext,
     },
   });
 
+  const submitMainForm = form.handleSubmit(async (value) => {
+    if (loading) return;
+    setError(null);
+    setLoading(true);
+    try {
+      const encodedParams = new URLSearchParams();
+      for (const key in value) {
+        const v = value[key];
+        if (v && typeof v === "object") {
+          encodedParams.append(key, JSON.stringify(value[key]));
+        } else {
+          encodedParams.append(key, value[key]);
+        }
+      }
+      const resp = await fetch(`${LICENSE_KEY_API_URL}/trial`, {
+        method: "POST",
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/x-www-form-urlencoded",
+        },
+        body: encodedParams,
+      });
+      if (resp?.status === 200) {
+        setSubmitState("send key success");
+        setLoading(false);
+        setCloseCta("Close");
+        track("Generate trial license", value);
```

Comment:
Same as above.  We have to be very careful about which properties we track.  The way you have it now, it will send the person's email and name.  We can't have our telemetry data contain any PII.  I would stick with just `accountPlan`, `source`, and `qty`.